sdk/armcore/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,105 @@
+
+### Customized Policy
+
+Because of adopting Azure Core which is a shared library across all Azure SDKs, there is also a minor change regarding how customized policy in configured.
+
+In old version (`services/**/mgmt/**`), we use the `(autorest.Client).Sender`, `(autorest.Client).RequestInspector` and `(autorest.Client).ResponseInspector` properties in `github.com/Azure/go-autorest/autorest` module to provide customized interceptor for the HTTP traffic.
+
+In new version (`sdk/**/arm**`), we use `(armcore.ConnectionOptions).PerCallPolicies` and `(armcore.ConnectionOptions).PerRetryPolicies` in `github.com/Azure/azure-sdk-for-go/sdk/armcore` module instead to inject customized policy to the pipeline.
+
+### Custom HTTP Client
+
+Similar to the customized policy, there are changes regarding how the custom HTTP client is configured as well. You can now use the `(armcore.ConnectionOptions).HTTPClient` option in `github.com/Azure/azure-sdk-for-go/sdk/armcore` module to use your own implementation of HTTP client and plug in what they need into the configuration.
+
+**In old version (`services/**/mgmt/**`)** you cannot customize the HTTP client.
+
+**In new version (`sdk/**/arm**`)**
+
+```go
+httpClient := NewYourOwnHTTPClient{}
+conn := armcore.NewConnection(credential, &armcore.ConnectionOptions{
+    HTTPClient: &httpClient,
+})
+```
+
+### Error Handling
+
+There is a minor namespace change in the exception class. To be specific, the errors returned by the SDK now is always of type `runtime.ResponseError` in `github.com/Azure/azure-sdk-for-go/sdk/internal/runtime` package which implements the `HTTPResponse` interface and `NonRetriableError` interface from `azcore` package.
+
+When there is an error in the SDK request, in the old version (`services/**/mgmt/**`), the return value will all be non-nil, and you can get the raw HTTP response from the response value. In the new version (`sdk/**/arm**`), the first return value will be empty and you need to cast the error to `HTTPResponse` interface to get the raw HTTP response.
+
+**In old version (`services/**/mgmt/**`)**
+
+```go
+resp, err := resourceGroupsClient.CreateOrUpdate(context.Background(), resourceGroupName, resourceGroupParameters)
+if err != nil {
+	log.Printf("Response code: %d", resp.Response.Response.StatusCode)
+}
+```
+
+**Equivalent in new version (`sdk/**/arm**`)**
+
+```go
+resp, err := resourceGroupsClient.CreateOrUpdate(context.Background(), resourceGroupName, resourceGroupParameters, nil)
+if err != nil {
+	rawResponse := err.(azcore.HTTPResponse).RawResponse()
+	log.Printf("Response code: %d", rawResponse.StatusCode)
+}
+```
+
+### Pagination
+
+In old version, `**ResultPage` is returned. It is a struct with some paging methods but no interfaces are defined regarding that.
+
+In new version, `**Pager` interface is returned. It is an extension of the interface `azcore.Pager`.
+
+## Need help?
```

Comment:
Is there any pattern change regarding LRO?

Reply:
Indeed we have some changes. I will add a section about this